src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,54 +1,116 @@
-import React from 'react';
+import React, { ChangeEvent, useCallback, useState } from 'react';
+import { getMovie } from '../../api';
 import './FindMovie.scss';
 
-// import { MovieCard } from '../MovieCard';
-
-export const FindMovie: React.FC = () => (
-  <>
-    <form className="find-movie">
-      <div className="field">
-        <label className="label" htmlFor="movie-title">
-          Movie title
-        </label>
-
-        <div className="control">
-          <input
-            type="text"
-            id="movie-title"
-            placeholder="Enter a title to search"
-            className="input is-danger"
-          />
-        </div>
+import { MovieCard } from '../MovieCard';
 
-        <p className="help is-danger">
-          Can&apos;t find a movie with such a title
-        </p>
-      </div>
+enum Error {
+  noError = '',
+  noSuchMovieError = 'Can\t find a movie with such title',
+  noMovieError = 'Select a movie first',
+  emptyQueryError = 'Enter search text',
+  repeatedMovieError = 'This movie already is in the list',
+}
+
+interface Props {
+  isMovieInTheList: (movie: Movie) => boolean;
+  addMovie: (movie: Movie) => void;
+}
+
+export const FindMovie: React.FC<Props> = ({ isMovieInTheList, addMovie }) => {
+  const [query, setQuery] = useState('');
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [errorMessage, setErrorMessage] = useState<Error>(Error.noError);
+
+  const findMovie = useCallback(async () => {
+    if (!query.trim().length) {
+      setErrorMessage(Error.emptyQueryError);
+    } else {
+      setErrorMessage(Error.noError);
+      const request = await getMovie(query);
+
+      if (request.Response === 'True') {
+        setMovie(request);
+        setErrorMessage(Error.noError);
+      } else {
+        setErrorMessage(Error.noSuchMovieError);
+      }
```

Comment:
Good, but such `if ... else ...` make your code more complicated to read and maintain